quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/SelectiveMigratorFunction.java

```diff
@@ -182,15 +183,25 @@ private static MigratorFunction createInsertMigrator(NullRecords nullRecords, Re
 				.map(column -> functionParameterMapping.get(column.getName()) + " " + column.getType().toString())
 				.collect(Collectors.toList());
 
-		TableRef sourceRef = refLog.getTableRefById(source.getName());
-		TableRef targetRef = refLog.getTableRefById(target.getName());
+		Multimap<TableRef, TableRef> tableMapping = refLog.getTableMapping(from, to);
+
+		TableRef sourceRef = tableMapping.keySet().stream()
+				.filter(tableRef -> tableRef.getRefId().equals(source.getName()))
+				.findFirst().get();
+
+		TableRef targetRef = tableMapping.get(sourceRef).stream()
+				.filter(tableRef -> tableRef.getRefId().equals(target.getName()))
+				.findFirst().get();
+
 		Map<ColumnRef, ColumnRef> columnMapping = refLog.getColumnMapping(sourceRef, targetRef);
 
 		Map<String, String> values = columnMapping.entrySet().stream()
-				.filter(entry -> columns.contains(entry.getKey().getName()))
+				.filter(entry -> columns.contains(entry.getValue().getName()))
```

Comment:
columns are the columns in the target table, not the source table, you have to check if the value is in. This caused a problem when you altered a column's name, it would not migrate.

Reply:
This bug costs me a whole day of debugging haha

Reply:
Not laughing at you, but this did make me chuckle. All too familiar with that feeling. 😄